azmq/detail/config/thread.hpp

```diff
@@ -0,0 +1,26 @@
+/*
+    Copyright (c) 2013-2014 Contributors as noted in the AUTHORS file
+
+    This file is part of azmq
+
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+*/
+#ifndef AZMQ_DETAIL_CONFIG_THREAD_HPP_
+#define AZMQ_DETAIL_CONFIG_THREAD_HPP_
+
+#if !defined(AZMQ_HAS_STD_THREAD)
```

Comment:
"To control main scenario I propose a define AZMQ_USE_BOOST_ASIO."

To me, it seems this should be the default case given the starting point of ONLY supporting Boost Asio. Maybe AZMQ_USE_STANDALONE_ASIO to control the main scenario, so -

#if !defined(AZMQ_USE_STANDALONG_ASIO)
    #define // default stuff here
#else 
    #define // Asio standalone specifics
#endif